src/main/java/org/reaktivity/nukleus/http_cache/internal/proxy/cache/CacheEntry.java

```diff
@@ -165,7 +165,13 @@ private void handleEndOfStream(
     public void serveClient(
             AnswerableByCacheRequest streamCorrelation)
     {
-        sendResponseToClient(streamCorrelation, true);
+        switch (this.state)
+        {
+            case PURGED:
+                throw new IllegalStateException("Can not serve client when entry is purged");
+            default:
+                sendResponseToClient(streamCorrelation, true);
```

Comment:
Remember to put `break` even though `default` case.

Reply:
this is fixed, see: https://github.com/reaktivity/nukleus-http-cache.java/pull/44/commits/0e903c2ebddd73a6f218e0c137f4e07290d1ef8f